src/Raven.Server/Utils/Enumerators/PulsedTransactionEnumerator.cs

```diff
@@ -44,9 +44,6 @@ public bool MoveNext()
                 _innerEnumerator = _getEnumerator != null ? _getEnumerator(_state) : _getEnumerable(_state).GetEnumerator();
             }
 
-            if (Current is Document doc)
```

Comment:
Who else is using `PulsedTransactionEnumerator'?

Reply:
Don't we need it there as well?

Reply:
`ForgetAbout()` is applicable only for compressed documents / revisions so we need it only for `Document` type. That is where I added `TransactionForgetAboutDocumentEnumerator` now instead of `if (Current is Document doc) ...` check which was a bit tricky. 

I prefer to have it explicitly stated in the code that during the enumeration we'll be also forgetting about decompression buffers. Note that this is not always what we want e.g. in replication we can't call `ForgetAbout()` _during_ the enumeration because we collect iterated items first (in `SortedList`) and operate on them.